src/plugins/data/public/search/fetch/handle_response.tsx

```diff
@@ -24,44 +24,34 @@ export function handleResponse(
 ) {
   const { rawResponse } = response;
 
-  if (rawResponse.timed_out) {
-    getNotifications().toasts.addWarning({
-      title: i18n.translate('data.search.searchSource.fetch.requestTimedOutNotificationMessage', {
-        defaultMessage: 'Data might be incomplete because your request timed out',
-      }),
-    });
-  }
+  // display warning toast notifications for timeouts and/or shard failures
+  const warnings = extractWarnings(rawResponse);
 
-  if (rawResponse._shards && rawResponse._shards.failed && !disableShardFailureWarning) {
-    const title = i18n.translate('data.search.searchSource.fetch.shardsFailedNotificationMessage', {
-      defaultMessage: '{shardsFailed} of {shardsTotal} shards failed',
-      values: {
-        shardsFailed: rawResponse._shards.failed,
-        shardsTotal: rawResponse._shards.total,
-      },
-    });
-    const description = i18n.translate(
-      'data.search.searchSource.fetch.shardsFailedNotificationDescription',
-      {
-        defaultMessage: 'The data you are seeing might be incomplete or wrong.',
-      }
-    );
+  const timedOut = warnings.find((w) => w.isTimeout === true);
```

Comment:
what about something like:

```
warnings.forEach(warning => {
  toasts.addWarning({ title: warning.title, description: warning.description })
```

Reply:
We still need to leverage `disableShardFailureWarning`, because this function is called from our `onResponse`. The goal behind the code here is to use that flag only suppress shard failures warnings, not timeout warnings. I think that it's good to stick to that, to not have unintentional differences.

Reply:
we can move that check into searchsource directly. this handleResponse method is confusing imo and is doing too much.